daemon/routes/iprule/iprule.go

```diff
@@ -80,16 +81,72 @@ func (r *Router) SetupRoutingRules(
 			return err
 		}
 
-		otherRuleID, err := calculateRulePriority(ipv6)
-		if err != nil {
-			return err
+		// PeerA (LAN-a 192.168.1.x) connects to PeerB (LAN-b 192.168.1.x)
+		// if PeerB allows its LAN access when used as Exit node
+		// then PeerB LAN access is the priority over PeerA LAN
+		if !routeToRemoteLan {
+			otherRuleID, err := calculateRulePriority(ipv6)
+			if err != nil {
+				return err
+			}
+
+			if err = addSuppressprefixLengthRule(
+				otherRuleID,
+				ipv6,
+			); err != nil {
+				return err
+			}
 		}
+	}
 
-		if err = addSuppressprefixLengthRule(
-			otherRuleID,
-			ipv6,
-		); err != nil {
-			return err
+	return nil
+}
+
+// PreventLocalTraffic adjust policy based routing rules
+func (r *Router) PreventLocalTraffic(
```

Comment:
Subjective, but I would rather call this `EnableLocalTraffic`, that way it's more clear that it's dependent on a bool argument whether the traffic is enabled and disabled. And even then, since it has two bool arguments it's impossible to tell whether `EnableLocalTraffic(interface, false, true)` would enable it or disable it. 

Another way to deal with bad readability of bool arguments is to simply use two public functions: EnableLocalTraffic and DisableLocalTraffic (both these functions would then just call the same private function setLocalTraffic) - this way it would be totally clear what is happening. Of course calling code would get more verbose because the `if routeToRemoteLan` condition would become the responsibility of the caller, but I think that is still better for readability.